File: Sammlung/barbaren.cpp
#include <iostream>
using namespace std;

main()
{
int Barbaren, Griechen, Personen, i;

Personen = 0;
for (Barbaren = 1; Barbaren < 7 ; Barbaren++)
	 {
		Griechen = Barbaren * 2;
		  if ((Barbaren+Griechen) % 4 == 0)
			  {
					 Personen = Barbaren + Griechen;
					 cout << "Anzahl der Personen: " << Personen << endl;
					 cout << "Anzahl der Barbaren: " << Barbaren << endl;
					 cout << "Anzahl der Griechen: " << Griechen << endl;
					 cin >>i;
				}
	  }
if (Personen == 0)
	cout << "Keine Loesung moeglich\n";

	cin >> i;
}

File: Sammlung/barbaren.pas
program barbar;
var Barbaren, Griechen, Personen, i: integer;

begin
Personen := 0;
for Barbaren := 1 to 6 do
    begin
        Griechen := Barbaren * 2;
        if (Barbaren+Griechen) MOD 4 = 0
           then
              begin
                Personen := Barbaren + Griechen;
                writeln('Anzahl der Personen: ',Personen);
                writeln('Anzahl der Barbaren: ',Barbaren);
                writeln('Anzahl der Griechen: ',Griechen);
                readln(i);
               end;
    end;
if Personen = 0
   then
      writeln('Keine Loesung moeglich');
readln(i);
end.
